src/components/Modals/Send/hooks/useSendDetails/useSendDetails.tsx

```diff
@@ -264,38 +264,49 @@ export const useSendDetails = (): UseSendDetailsReturnType => {
 
   // eslint-disable-next-line react-hooks/exhaustive-deps
   const handleInputChange = useCallback(
-    debounce(async (inputValue: string) => {
-      setValue(SendFormFields.SendMax, false)
-      const key =
-        fieldName !== SendFormFields.FiatAmount
-          ? SendFormFields.FiatAmount
-          : SendFormFields.CryptoAmount
-      const amount =
-        fieldName === SendFormFields.FiatAmount
-          ? bnOrZero(bn(inputValue).div(price)).toString()
-          : bnOrZero(bn(inputValue).times(price)).toString()
-
-      setValue(key, amount)
+    debounce(
+      async (inputValue: string) => {
+        setLoading(true)
+        setValue(SendFormFields.SendMax, false)
+        const key =
+          fieldName !== SendFormFields.FiatAmount
+            ? SendFormFields.FiatAmount
+            : SendFormFields.CryptoAmount
+        const amount =
+          fieldName === SendFormFields.FiatAmount
+            ? bnOrZero(bn(inputValue).div(price)).toString()
+            : bnOrZero(bn(inputValue).times(price)).toString()
+
+        setValue(key, amount)
+
+        const estimatedFees = await estimateFormFees().catch(e => {
+          setValue(SendFormFields.AmountFieldError, 'common.insufficientFunds')
+          setLoading(false)
+
+          throw e
+        })
+        setValue(SendFormFields.EstimatedFees, estimatedFees)
```

Comment:
Is the intent we call `setValue` here even if we failed to initialize `estimatedFees`?

Reply:
Not exactly, it will set `undefined`, which is not an issue since user will not be able to continue the flow because of the error state.

We could wrap this in an if statement, but setting this has no cost so I think it makes sense not to introduce an additional branch!

Reply:
Since we're now using try/catch, moved the `setValue()` call inside it.